src/transaction/deploy_account.rs

```diff
@@ -204,12 +204,19 @@ impl DeployAccount {
         self.handle_nonce(state)?;
 
         let mut transactional_state = state.create_transactional()?;
-        let mut tx_exec_info = self.apply(
+        let original_state = state.create_transactional()?;
+        let mut tx_exec_info = match self.apply(
             &mut transactional_state,
             block_context,
             #[cfg(feature = "cairo-native")]
             program_cache.clone(),
-        )?;
+        ) {
+            Ok(info) => info,
+            Err(e) => {
+                drop(original_state);
```

Comment:
Doesn't `drop` get called implicitly when returning the error?

Reply:
It didn't let me do it implicitly that's why I added it

Reply:
I modified the function, i remove the drop by encapsulating it